### PR TITLE
fix: Cannot read property 'startsWith' of undefined

### LINKED ISSUE #1356: Cannot read property 'level' of undefined
#### Steps to reproduce

Run the following README.md:

```README.md
# Heading 1

Test

## Heading 2

### Heading 3A

Lorem ipsum

#### Heading 4A

##### Heading 5A

Lorem ipsum

### Heading 3B

Lorem ipsum

#### Heading 4B <!-- {docsify-ignore} -->

##### Heading 5B

Lorem ipsum
```

#### What is current behaviour

The sidebar does not show up and the console shows:

```

```docsify.min.js:1 Uncaught TypeError: Cannot read property 'level' of undefined
    at $n.sidebar (docsify.min.js:1)
    at Vn.Xn._renderSidebar (docsify.min.js:1)
    at Vn.Pn (docsify.min.js:1)
    at docsify.min.js:1
    at o (docsify.min.js:1)
    at B (docsify.min.js:1)
    at t (docsify.min.js:1)
    at docsify.min.js:1
    at docsify.min.js:1
    at docsify.min.js:1
```

#### What is the expected behaviour

Sidebar shows up

#### Other relevant information

- [x] Bug does still occur when all/other plugins are disabled?

- Your OS: macOS Catalina
- Node.js version: 12.18.3
- npm/yarn version: 6.14.6
- Browser version: Chrome Version 84.0.4147.135 (Official Build) (64-bit)
- Docsify version: 4.11.6
- Docsify plugins: none

#### Please create a reproducible sandbox 

[Code Sandbox](https://codesandbox.io/s/unruffled-cdn-16042)

### PR DESCRIPTION
**Pull request by sy-records**
**Summary**

I fixed `#1356` and found another error.

image

```js
    const path = paths[0];

    if (Array.isArray(config.pathNamespaces)) {
      namespaceSuffix =
        config.pathNamespaces.find(prefix => path.startsWith(prefix)) ||
        namespaceSuffix;
    } 
```

`path` is `undefined` when only a readme file is present.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Repo settings
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

### CODE REVIEW

**Comment by sy-records** [comment]
@palkan Can you review it?

**Review by anikethsaha — APPROVED** [review]

File: src/plugins/search/search.js [hunk 1]
```diff
@@ -208,10 +208,10 @@ export function init(config, vm) {
 
     if (Array.isArray(config.pathNamespaces)) {
       namespaceSuffix =
-        config.pathNamespaces.find(prefix => path.startsWith(prefix)) ||
+        config.pathNamespaces.find(prefix => path && path.startsWith(prefix)) ||
```
Comment (palkan) [inline src/plugins/search/search.js:211]: What about checking for the `paths.length` here: https://github.com/docsifyjs/docsify/pull/1358/files#diff-4ce3780eb5b2c8eb805df4e0c950cd7dR206 ?

If I understood correctly, the problem occurs only when there is nothing to index, i.e., `paths == []`. If so, adding a `paths.length` would be more descriptive.

**Review by palkan — CHANGES_REQUESTED** [review]

**Review by Koooooo-7 — COMMENTED** [review]
I think @palkan said make sense, 
and you can add ` path && path.length && path.startsWith` also. :sweat_smile:

File: src/plugins/search/search.js (hunk 1 above)
Comment (anikethsaha) [reply src/plugins/search/search.js:211]: Are there any cases where `path` will be an array? I think there are non, cause if so I think checking length makes sense.
Also, arrays don't contain the method `startsWith`. 

Let me know if I am missing something here.

Comment (palkan) [reply src/plugins/search/search.js:211]: I meant `paths`, not `path`. See line 201 (and we can add an outer `if (paths.length) { ... }` clause before line 206).

Comment (sy-records) [reply src/plugins/search/search.js]: done.

@anikethsaha An array if _sidebar exists, otherwise empty.

Comment (anikethsaha) [reply src/plugins/search/search.js]: COol

**Review by palkan — APPROVED** [review]
LGTM

**Review by anikethsaha — APPROVED** [review]

**Review by Koooooo-7 — APPROVED** [review]

**Pull request merged**